### PR TITLE
Quiet gcc compiler complaint

### PR DESCRIPTION
**Pull request by JetstreamRoySprowl**
gcc would like to see the extra paren to clarify logical precedence

### CODE REVIEW

File: inc/umock_c_internal.h [hunk 1]
```diff
@@ -147,8 +147,8 @@ typedef int(*TRACK_DESTROY_FUNC_TYPE)(PAIRED_HANDLES* paired_handles, const void
...
-    if (result && \
-        ((typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].bytes != NULL) && (memcmp(*((void**)&typed_right->arg_name), typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].bytes, typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].length) != 0)) \
+    if ((result && \
+        ((typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].bytes != NULL) && (memcmp(*((void**)&typed_right->arg_name), typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].bytes, typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].length) != 0))) \
         || ((typed_right->validate_arg_buffers[COUNT_OF(typed_right->validate_arg_buffers) - DIV2(count)].bytes != NULL) && (memcmp(*((void**)&typed_left->arg_name), typed_right->validate_arg_buffers[COUNT_OF(typed_right->validate_arg_buffers) - DIV2(count)].bytes, typed_right->validate_arg_buffers[COUNT_OF(typed_right->validate_arg_buffers) - DIV2(count)].length) != 0)) \
         ) \
     { \
```
Comment (dcristoloveanu) [inline inc/umock_c_internal.h]: This should be 

(result &&

(
(A ...)
|| (B ...)
)

)

The intent is to evaluate the arg checks only if result is 1

The way you wrote it the part after || will be evaluated even if result is 0, which I think is incorrect.

**Comment by JetstreamRoySprowl** [comment]
Thank you very much for finding that, Dan. It would have been ugly,

**Comment by JetstreamRoySprowl** [comment]
Requested change done.

**Comment by dcristoloveanu** [comment]
This is what I see in the code:

    if (result && \
        (((typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].bytes != NULL) && (memcmp(*((void**)&typed_right->arg_name), typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].bytes, typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].length) != 0))) \
        || ((typed_right->validate_arg_buffers[COUNT_OF(typed_right->validate_arg_buffers) - DIV2(count)].bytes != NULL) && (memcmp(*((void**)&typed_left->arg_name), typed_right->validate_arg_buffers[COUNT_OF(typed_right->validate_arg_buffers) - DIV2(count)].bytes, typed_right->validate_arg_buffers[COUNT_OF(typed_right->validate_arg_buffers) - DIV2(count)].length) != 0)) \
        ) \

I think it has to be 

    if (result && \
        (((typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].bytes != NULL) && (memcmp(*((void**)&typed_right->arg_name), typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].bytes, typed_left->validate_arg_buffers[COUNT_OF(typed_left->validate_arg_buffers) - DIV2(count)].length) != 0)) \
        || ((typed_right->validate_arg_buffers[COUNT_OF(typed_right->validate_arg_buffers) - DIV2(count)].bytes != NULL) && (memcmp(*((void**)&typed_left->arg_name), typed_right->validate_arg_buffers[COUNT_OF(typed_right->validate_arg_buffers) - DIV2(count)].bytes, typed_right->validate_arg_buffers[COUNT_OF(typed_right->validate_arg_buffers) - DIV2(count)].length) != 0))) \
        ) \

Basically the () should go around the 2 expressions that are OR'd.

Let me know if I am reading this wrong.

---
In reply to: [385475503](https://github.com/Azure/umock-c/pull/36#issuecomment-385475503) [](ancestors = 385475503)

**Comment by JetstreamRoySprowl** [comment]
I am having a very bad day, and my compiler is lying to me. It can only get worse if Tamer realizes that I can be replaced by a script that makes random code changes and submits them to you as pull requests.

I have updated the branch again (sigh).

**Comment by dcristoloveanu** [comment]
LOL. Had a good giggle at the random code change script.
Don’t worry about it 😊. Thanks!

Cheers,
/Dan
